beam_jobs/base_model_validator.py

```diff
@@ -0,0 +1,239 @@
+ERROR_CATEGORY_NAME_CHECK = 'name check'
+ERROR_CATEGORY_OUTPUT_CHECK = 'output check'
+ERROR_CATEGORY_PRIVATE_COMMIT_CHECK = 'post commit is private check'
+ERROR_CATEGORY_PROPERTY_FETCH_CHECK = 'fetch properties'
+ERROR_CATEGORY_RATED_ON_CHECK = 'rated on check'
+ERROR_CATEGORY_RATINGS_CHECK = 'ratings check'
+ERROR_CATEGORY_REFERENCE_CHECK = 'reference check'
+ERROR_CATEGORY_AUTHOR_CHECK = 'author check'
+ERROR_CATEGORY_REVIEWER_CHECK = 'reviewer check'
+ERROR_CATEGORY_STATE_NAME_CHECK = 'state name check'
+ERROR_CATEGORY_SUMMARY_CHECK = 'summary check'
+ERROR_CATEGORY_TIME_FIELD_CHECK = 'time field relation check'
+ERROR_CATEGORY_TYPE_CHECK = 'type check'
+ERROR_CATEGORY_VERSION_CHECK = 'version check'
+ERROR_CATEGORY_STALE_CHECK = 'stale check'
+ERROR_CATEGORY_INVALID_IDS_IN_FIELD = 'invalid ids in field'
+
+VALIDATION_MODE_NEUTRAL = 'neutral'
+VALIDATION_MODE_STRICT = 'strict'
+VALIDATION_MODE_NON_STRICT = 'non-strict'
+
+
+class ValidateModelIdWithRegex(beam.DoFn):
+    """DoFn to validate model ids against a given regex string."""
+
+    def __init__(self, regex_string):
+        """Initializes the ValidateModelIdWithRegex DoFn.
+
+        Args:
+          regex_string: str. A regex pattern for valid ids.
+        """
+        super(init)
+        self.regex_string = regex_string
+
+    def process(self, model):
+        """Function that defines how to process each element in a pipeline of
+        models.
+
+        Args:
+          model: datastore_services.Model. Entity to validate.
+
+        Yields:
+          beam.pvalue.TaggedOutput. An element of the output PCollection for
+          the doFn which represents an error as a key value pair.
+        """
+        element = model.clone()
+        regex_string = self.regex_string
+
+        if not re.compile(regex_string).match(element.id):
+            yield beam.pvalue.TaggedOutput('error_category_id_check', (
+                'model %s' % ERROR_CATEGORY_ID_CHECK,
+                'Entity id %s: Entity id does not match regex pattern' % (
+                    element.id)
+            ))
+        yield element
+
+
+class ValidateDeleted(beam.DoFn):
+    """DoFn to check whether models marked for deletion are stale."""
+
+    def process(self, model):
+        """Function that defines how to process each element in a pipeline of
+        models.
+
+        Args:
+          model: datastore_services.Model. Entity to validate.
+
+        Yields:
+          beam.pvalue.TaggedOutput. An element of the output PCollection for
+          the doFn which represents an error as a key value pair.
+        """
+        element = model.clone()
+        date_now = datetime.datetime.utcnow()
+
+        date_before_which_models_should_be_deleted = (
+            date_now -
+            cron_services.PERIOD_TO_HARD_DELETE_MODELS_MARKED_AS_DELETED
+        )
+
+        period_to_hard_delete_models_in_days = (
+            cron_services.PERIOD_TO_HARD_DELETE_MODELS_MARKED_AS_DELETED.days)
```

Comment:
Hmm, I think it may be better to move this constant into this file. WDYT @seanlip ?

Reply:
I don't have a strong opinion. Check with @vojtechjelinek?

Reply:
Let's keep it in `cron_services` for now, because I think we should limit the importing of jobs in services.